server/src/main/resources/logback.xml

```diff
@@ -20,7 +20,7 @@
     <appender name="CandlepinAppender" class="ch.qos.logback.core.rolling.RollingFileAppender">
         <file>${LOG_DIR}/candlepin.log</file>
         <encoder>
-            <pattern>%d{ISO8601} [thread=%thread] [%X{requestType}=%X{requestUuid}, org=%X{org}, csid=%X{csid}] %-5p %c - %m%n</pattern>
+            <pattern>%d{ISO8601} [thread=%thread] [%X{requestType}=%X{requestUuid}, %X{requestSubType}=%X{requestName}, org=%X{org}, csid=%X{csid}] %-5p %c - %m%n</pattern>
```

Comment:
This is going to result in a floating "=, " for the majority of log entries. If we're going to use this method for omitting the entry when it's not present, we also need to do that for the separators as well, which is a bit hack.

We should look for a better option here, such that we can avoid printing the whole thing if the target variable is empty. All else failing, we can update the logging format here to specify the value separator and the keypair separator, and then inject them ourselves as suggested above.

Reply:
Looks like ```%equals{job_key=%X{jobKey}, }{job_key=, }{}``` would omit the job key printing if the value is not present in the MDC. This would also allow us to drop the request name bit as well, unless there's some other intended use for that.

Reply:
Scratch that; wrong logging format. This works, however:
```%replace(job_key=%X{jobKey}, ){'job_key=, ', ''}```

Reply:
Replacement is working correctly for job key field. 
Leaving other as it is (`org=, csid=`).

PS: Even a slightest mistake in Logback patterns will make your logs disappears. Learnt this the hard way :(

Reply:
Yeah, leaving the others is fine for now. I'm poking people on the value of conditionally removing them, and we can make a decision on that front later.